### PR TITLE
Feature/predefined autoscale formula

### PR DESCRIPTION
**Pull request by johnnyzhang82**
@timotheeguerin Added sample autoscale formulas.

### CODE REVIEW

File: data/autoscale-formula/AccountingForParallelTasks.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Determine whether 70 percent of the samples have been recorded in the past 15 minutes; if not, use last sample
+$samples = $ActiveTasks.GetSamplePercent(TimeInterval_Minute * 15);
+$tasks = $samples < 70 ? max(0,$ActiveTasks.GetSample(1)) : max( $ActiveTasks.GetSample(1),avg($ActiveTasks.GetSample(TimeInterval_Minute * 15)));
+// Set the number of nodes to add to one-fourth the number of active tasks (the MaxTasksPerComputeNode property on this pool is set to 4, adjust this number for your use case)
```
Comment (timotheeguerin) [inline data/autoscale-formula/AccountingForParallelTasks.txt]: Maybe add a spec before the comments on this file

Comment (timotheeguerin) [reply data/autoscale-formula/AccountingForParallelTasks.txt]: Also here is says 1/4 of the number of active task. How about we have a variable at the top for this.

File: data/autoscale-formula/SettingInitialPoolSize.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+$TargetDedicated = 4;
```
Comment (timotheeguerin) [inline data/autoscale-formula/SettingInitialPoolSize.txt]: Add a comment here maybe saying 4 is the initial size

File: data/autoscale-formula/TaskBasedAdjustment.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Get pending tasks for the past 15 minutes.
+$samples = $ActiveTasks.GetSamplePercent(TimeInterval_Minute * 15);
+// If we have fewer than 70 percent data points, we use the last sample point, otherwise we use the maximum of last sample point and the history average.
```
Comment (timotheeguerin) [inline data/autoscale-formula/TaskBasedAdjustment.txt:5]: Add some spaces before comments

File: data/autoscale-formula/TaskBasedAdjustment.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+$tasks = $samples < 70 ? max(0, $ActiveTasks.GetSample(1)) : max( $ActiveTasks.GetSample(1), avg($ActiveTasks.GetSample(TimeInterval_Minute * 15)));
+// If number of pending tasks is not 0, set targetVM to pending tasks, otherwise half of current dedicated.
+$targetVMs = $tasks > 0? $tasks:max(0, $TargetDedicated / 2);
+// The pool size is capped at 20, if target VM value is more than that, set it to 20. This value should be adjusted according to your use case.
+$TargetDedicated = max(0, min($targetVMs, 20));
```
Comment (timotheeguerin) [inline data/autoscale-formula/TaskBasedAdjustment.txt]: Could we have max and min as constant at the top of the file?

File: data/autoscale-formula/TimeBasedAdjustment.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+$curTime = time();
```
Comment (timotheeguerin) [inline data/autoscale-formula/TimeBasedAdjustment.txt:4]: Just add a top comment describing briefly what this forumla does(Might be good to have that for all of them)

File: data/autoscale-formula/SettingInitialPoolSize.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+lifespan = time() - time(\"Mon, 06 Oct 2014 10:20:00 GMT\");
```
Comment (johnnyzhang82) [inline data/autoscale-formula/SettingInitialPoolSize.txt]: "Mon, 06 Oct 2014 10:20:00 GMT" is supposed to indicate datetime now. I think we should mention that here too. @timotheeguerin

Comment (timotheeguerin) [reply data/autoscale-formula/SettingInitialPoolSize.txt]: yeah good idea

File: data/autoscale-formula/TimeBasedAdjustment.txt (hunk 5 above)
Comment (johnnyzhang82) [reply data/autoscale-formula/TimeBasedAdjustment.txt:4]: Added.

File: data/autoscale-formula/SettingInitialPoolSize.txt (hunk 2 above)
Comment (johnnyzhang82) [reply data/autoscale-formula/SettingInitialPoolSize.txt]: Added.

File: data/autoscale-formula/TaskBasedAdjustment.txt (hunk 3 above)
Comment (johnnyzhang82) [reply data/autoscale-formula/TaskBasedAdjustment.txt:5]: Added a brief description at top.

File: data/autoscale-formula/TaskBasedAdjustment.txt (hunk 4 above)
Comment (johnnyzhang82) [reply data/autoscale-formula/TaskBasedAdjustment.txt]: min and max are math function here. You mean some values?

Comment (timotheeguerin) [reply data/autoscale-formula/TaskBasedAdjustment.txt]: yeah at the top

```
min = 0      // Minimum number of nodes
max = 20   // Maximum number of nodes
```

**Pull request merged**